### PR TITLE
Getting ride of guava

### PR DESCRIPTION
**Pull request by velo**
Got class def not found when running with maven 3.5.4

Since it was not critical, I decided to take it away

### CODE REVIEW

File: src/main/java/io/takari/maven/builder/smart/ProjectComparator.java [hunk 1]
```diff
@@ -38,7 +36,7 @@
 
   public static Comparator<MavenProject> create(MavenSession session) {
     final ProjectDependencyGraph dependencyGraph = session.getProjectDependencyGraph();
-    return create0(DependencyGraph.fromMaven(dependencyGraph), ImmutableMap.of(), p -> id(p));
+    return create0(DependencyGraph.fromMaven(dependencyGraph), new HashMap<>(), p -> id(p));
```
Comment (ifedorenko) [inline src/main/java/io/takari/maven/builder/smart/ProjectComparator.java]: I think `emptyMap` is more appropriate here

File: src/main/java/io/takari/maven/builder/smart/ReactorBuildQueue.java [hunk 2]
```diff
@@ -43,8 +41,8 @@ public ReactorBuildQueue(Collection<MavenProject> projects,
       }
     }
 
-    this.rootProjects = ImmutableSet.copyOf(rootProjects);
-    this.projects = ImmutableSet.copyOf(projects);
+    this.rootProjects = new HashSet<>(rootProjects);
+    this.projects = new HashSet<>(projects);
```
Comment (ifedorenko) [inline src/main/java/io/takari/maven/builder/smart/ReactorBuildQueue.java]: We need immutable ordered sets here.

File: src/main/java/io/takari/maven/builder/smart/ReactorBuildStats.java [hunk 3]
```diff
@@ -39,22 +38,22 @@
 
   private ReactorBuildStats(Map<String, AtomicLong> serviceTimes,
       Map<String, AtomicLong> bottleneckTimes) {
-    this.serviceTimes = ImmutableMap.copyOf(serviceTimes);
-    this.bottleneckTimes = ImmutableMap.copyOf(bottleneckTimes);
+    this.serviceTimes = new HashMap<>(serviceTimes);
+    this.bottleneckTimes = new HashMap<>(bottleneckTimes);
```
Comment (ifedorenko) [inline src/main/java/io/takari/maven/builder/smart/ReactorBuildStats.java]: We need immutable maps here, and I'd make them ordered just for consistency.

File: src/main/java/io/takari/maven/builder/smart/ReactorBuildStats.java [hunk 4]
```diff
@@ -39,22 +38,22 @@
...
 
   private static String projectGA(MavenProject project) {
     return project.getGroupId() + ":" + project.getArtifactId();
   }
 
   public static ReactorBuildStats create(Collection<MavenProject> projects) {
-    ImmutableMap.Builder<String, AtomicLong> serviceTimes = ImmutableMap.builder();
-    ImmutableMap.Builder<String, AtomicLong> bottleneckTimes = ImmutableMap.builder();
+    Map<String, AtomicLong> serviceTimes = new HashMap<>();
+    Map<String, AtomicLong> bottleneckTimes = new HashMap<>();
```
Comment (ifedorenko) [inline src/main/java/io/takari/maven/builder/smart/ReactorBuildStats.java]: We need immutable maps here, and I'd make them ordered just for consistency.

**Comment by velo** [comment]
Changes applied and CLA sent

**Comment by cstamas** [comment]
@velo Happy Birthday!

**Comment by gnodet** [comment]
Superseded with #26

**Pull request closed without merging**